iothub/service/samples/how to guides/AutomaticDeviceManagementSample/AutomaticDeviceManagementSample.cs

```diff
@@ -83,6 +83,7 @@ private static void CreateDeviceContent(Configuration configuration, string conf
 
         private static void CreateMetricsAndTargetCondition(Configuration configuration)
         {
+            configuration.Metrics = new ConfigurationMetrics();
```

Comment:
Added to prevent NullReferenceException

Reply:
```suggestion
            configuration.Metrics ??= new ConfigurationMetrics();
```

Reply:
Or consider fixing it upstream so it can't be null.